.github/workflows/docs.yml

```diff
@@ -0,0 +1,33 @@
+name: docs
+
+on:
+  pull_request:
+    paths:
+      - 'docs/**'
+
+# Cancel previous PR builds.
+concurrency:
+  # Cancel all workflow runs except latest within a concurrency group. This is achieved by defining a concurrency group for the PR.
+  # Non-PR builds have singleton concurrency groups.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  docs-checks:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version:
+          - 11
+          - 17
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: ${{ matrix.java-version }}
+      - name: Maven Checks
+        run: |
+          .github/bin/retry ./mvnw install -B --strict-checksums -V -T C1 -DskipTests -P ci -pl ':trino-docs'
```

Comment:
The challenge is that we don't track where what from docs is tested. For example I see `io.trino.security.TestFileBasedSystemAccessControl#testDocsExample` that would mean that we need to run tests for `trino-main`?

Reply:
Sure, let's run all of trino-main. Maybe trino-tests too, didn't check.

Reply:
I couldn't find docs tests in `trino-tests`. Going to add only `trino-main`.

Reply:
@findepi @kokosing Can we use testing `docs` group likes https://github.com/trinodb/trino/pull/9689/commits/22af62754523d99575e336a6d57a6591dd0148ad to reduce build time?

Reply:
Looks good to me

Reply:
The docs profile looks useful but it's possible we may miss adding the groups (and hence test coverage?).

Even if we run all tests in those 3 modules it'd still save time compared to what we have today (and we can iterate - the idea to have docs tests in docs module makes sense to me)?

Reply:
@hashhar this.

Reply:
I don't have strong opinion, but all approach have risk we may miss either.  Even if we run with module level or move tests to docs module, someone may add docs tests to other modules.

Reply:
@ebyhr in the worst case, we will notice failure on master and fixed the CI jobs, or move the test.

is there a set of modules we all agree to include here?
`trino-main` and `trino-tests`?

Reply:
@findepi I meant the worst case is little hard to avoid and "fixed the CI jobs, or move the test" is almost same as adding to `docs` group (= requires manual process either).

As far as I confirmed, below modules have docs tests. I will send a new testing PR without docs module if we want to verify on CI. 
* trino-main
* trino-plugin-toolkit
* trino-resource-group-manager

Reply:
> As far as I confirmed, below modules have docs tests


thanks for the list!

let's add `trino-tests` (future-proofing) to the list, and I'm fine.

Reply:
@findepi Updated. The latest `docs.yml` was tested in #9689